controllers/component_controller.go

```diff
@@ -312,10 +314,15 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 					devfileBytes, devfileLocation, err = devfile.FindAndDownloadDevfile(gitURL)
 					if err != nil {
+						// Increment the import git repo failed metric on non-user errors
+						if _, ok := err.(*devfile.NoFileFound); !ok {
+							metrics.ImportGitRepoFailed.Inc()
+						}
 						log.Error(err, fmt.Sprintf("Unable to read the devfile from dir %s %v", gitURL, req.NamespacedName))
 						_ = r.SetCreateConditionAndUpdateCR(ctx, req, &component, err)
 						return ctrl.Result{}, err
 					}
+					metrics.ImportGitRepoSucceeded.Inc()
```

Comment:
Since a "success" scenario here also includes "user errors" (basically anything other than a system error, we also want to increment above when err != nil and the err is `devfile.NoFileFound`.

Reply:
updated